### PR TITLE
Define rust

### LINKED ISSUE #947: Document the theoretical Rust Language release thing
Basically, when there's a new rustc release v1.N, there's also a theoretical release of "The Rust Language v1.N" and that is what the reference documents.

### PR DESCRIPTION
**Pull request by Havvy**
Fixes #947 

I'm not super satisfied with what I've written here, but I think it's an improvement over not saying anything at all. I've also included a how to find other versions thing in there since it felt appropriate.

### CODE REVIEW

File: src/introduction.md [hunk 1]
```diff
@@ -1,102 +1,98 @@
...
 
+## Rust Releases
+
+Rust has a new language release every six weeks.
+The first stable release of the language was Rust 1.0.0, followed by Rust 1.1.0 and so on.
+Tools (`rustc`, `cargo`, etc.) and documentation ([Standard library], this book, etc.) are released with the language release.
+
+While tools often have the same version as the Rust language, the language exists as its own concept separately from them.
+
+The latest release of this book, matching the latest Rust version, can always be found at [https://doc.rust-lang.org/reference/](https://doc.rust-lang.org/reference/).
```
Comment (ehuss) [inline src/introduction.md:26]:
```suggestion
The latest release of this book, matching the latest Rust version, can always be found at <https://doc.rust-lang.org/reference/>.
```

File: src/introduction.md [hunk 2]
```diff
@@ -1,102 +1,98 @@
...
+Prior versions can be found by adding the Rust version before the "reference" directory.
+For example, the Reference for Rust 1.49.0 is located at [https://doc.rust-lang.org/1.49.0/reference/](https://doc.rust-lang.org/1.49.0/reference/).
```
Comment (ehuss) [inline src/introduction.md:28]:
```suggestion
For example, the Reference for Rust 1.49.0 is located at <https://doc.rust-lang.org/1.49.0/reference/>.
```

File: src/introduction.md [hunk 3]
```diff
@@ -1,102 +1,98 @@
...
 
 </div>
 
+## Rust Releases
+
+Rust has a new language release every six weeks.
+The first stable release of the language was Rust 1.0.0, followed by Rust 1.1.0 and so on.
+Tools (`rustc`, `cargo`, etc.) and documentation ([Standard library], this book, etc.) are released with the language release.
+
+While tools often have the same version as the Rust language, the language exists as its own concept separately from them.
```
Comment (ehuss) [inline src/introduction.md:24]: I'm not entirely clear what this sentence is trying to say.  Can you elaborate?  I can kinda see that "Rust the Language" and "`rustc` the compiler" are two separate things that happen to be versioned together, but I'm uncertain if it needs to be called out. I think the previous sentence makes it clear how the tools are packaged and versioned.

Comment (Havvy) [reply src/introduction.md:24]: I wrote this before I wrote what was before it, and yeah, it's trying to say exactly that. It's important more so if there's another compiler that gets mainstream usage, but I can just remove it for now. The previous paragraph explains it clearly enough.

**Review by ehuss — APPROVED** [review]
👍

**Pull request merged**